### PR TITLE
Button should support Icons from pcln-icons

### PR DESCRIPTION
**Pull request by sdalonzo**
- Upgrade to latest Rush (5.90.0) and PNPM (7.26.3)
- Use build cache for every project
- Upgrade Heft to latest (0.49.5)
- upgrade CodeQL to version 2. post-merge we can drop LGTM
- standardize on latest ESLint (8.33.0)
- bump dependencies in eslint-config
- used styledSystemPropTypes
- Add `IconLeft` and `IconRight` props to Button, expecting a pcln-icons component
  - stories with 2 icons and 1 icon, no text

### CODE REVIEW

**Review by scottned — APPROVED** [review]

**Review by craigpalermo — APPROVED** [review]

File: apps/docs/package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "version": "1.0.0",
   "scripts": {
     "start": "next -p 3030",
-    "lint": "next lint",
+    "lint": "",
```
[the remark this answers is not in the document]
Comment (sdalonzo) [reply apps/docs/package.json:7]: we are working to deprecate the next docsite by the end of Q1 so the linter isn't helping anything

File: packages/carousel/package.json [hunk 2]
```diff
@@ -71,13 +71,13 @@
...
     "typescript": "^4.5.4"
   },
   "peerDependencies": {
-    "react": ">= 16.10.0",
-    "react-dom": ">= 16.10.0",
-    "styled-components": ">=5 <5.3.4",
-    "pcln-design-system": "5.6.1"
+    "react": ">=16.10.0",
+    "react-dom": ">=16.10.0",
+    "styled-components": ">=5 <5.3.4 || >=5.3.7 <6",
```
Comment (pcln-james) [inline packages/carousel/package.json:80]: is this specific to this package or should it be added to all?

**Review by pcln-james — APPROVED** [review]
LGTM! I think just autocomplete is missing that specificity on styled-components peer but that might be on purpose

File: packages/carousel/package.json (hunk 2 above)
Comment (sdalonzo) [reply packages/carousel/package.json:80]: I added this because 5.3.7 removed the Ukraine messaging that was tagged as a supply chain attack.

**Pull request merged**